web/html/src/manager/recurring/recurring-actions-edit.tsx

```diff
@@ -0,0 +1,253 @@
+import * as React from "react";
+
+import _isEqual from "lodash/isEqual";
+
+import { Button } from "components/buttons";
+import { AsyncButton } from "components/buttons";
+import { Form, Select } from "components/input";
+import { Utils as MessagesUtils } from "components/messages";
+import { InnerPanel } from "components/panels/InnerPanel";
+import { RecurringEventPicker } from "components/picker/recurring-event-picker";
+import { StatesPicker } from "components/states-picker";
+import { Toggler } from "components/toggler";
+
+import Network from "utils/network";
+
+import { DisplayHighstate } from "../state/display-highstate";
+
+type Props = {
+  schedule?: any;
+  onActionChanged: (arg0: any) => any;
+  onSetMessages: (arg0: any) => any;
+  onError: (arg0: any) => any;
+};
+
+type State = {
+  minions?: any[];
+  active: boolean;
+  targetId?: any;
+  recurringActionId?: any;
+  scheduleName?: any;
+  type?: any;
+  targetType?: any;
+  actionTypeDescription?: any;
+  cron?: any;
+  details?: any;
+};
+
+class RecurringActionsEdit extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      minions: window.minions,
+      active: true,
+      details: {},
+    };
+
+    if (this.isEdit()) {
+      this.setSchedule(this.props.schedule);
+    } else {
+      this.getTargetType();
+    }
+  }
+
+  componentDidMount(): void {
+    if (this.props.schedule && this.props.schedule.recurringActionId) {
+      this.getDetailsData();
+    }
+  }
+
+  componentDidUpdate(prevProps: Readonly<Props>, prevState: Readonly<State>, snapshot?: any): void {
+    if (!_isEqual(prevProps.schedule, this.props.schedule)) {
+      this.getDetailsData();
+    }
+  }
+
+  getDetailsData(): void {
+    Network.get(`/rhn/manager/api/recurringactions/${this.props.schedule.recurringActionId}/details`)
+      .then((details) => {
+        this.setState({ details });
+      })
+      .catch(this.props.onError);
+  }
+
+  updateSchedule = (schedule) => {
+    return Network.post("/rhn/manager/api/recurringactions/save", schedule)
+      .then((_) => {
+        const successMsg = <span>{t("Schedule successfully" + (this.isEdit() ? " updated." : " created."))}</span>;
```

Comment:
I would recommend toggling this translation separately, e.g.

```tsx
{this.isEdit() ? t("Schedule successfully updated.") : t("Schedule successfully created.")}
```

That way it's easier for translators for languages that have a different word order.